wordpress-mobile/WordPress-Android
# My Site Dashboard: Tabs - Track events with tab source

Closes #16007 

This PR supports tracking the following events with the "tab_source" property:
`quick_action_tapped`
` my_site_pull_to_refresh`

When tabs are enabled and when the action is requested (PTR or Quick link button tap) an event is posted to `onTrackWithTabSource` which is observed in `MySiteFragment`. `MySiteFragment` then gets the current fragment and passes along the "tab source/current tab" via the `onTrackWithTabSource` fun. The request then makes it back to the VM which actually tracks the event.

**To test:**
**Test A: Tabs disabled**
- Launch the app
- Navigate to Me -> App Settings -> Debug Settings
- Disable `MySiteDashboardTabsFeatureConfig` and restart the app
- Navigate to My Site View
- Tap on the following (returning back to My Site View after each click):  
   - All Quick Link buttons (Stats, Pages, Posts, and Media) 
-  Swipe down on the My Site View to request a pull-to-refresh
- View the logs and note that "tab_source" is not included in the properties of the track event. 
```
🔵 Tracked: my_site_pull_to_refresh
🔵 Tracked: quick_action_tapped, Properties: {"button":"stats"}
🔵 Tracked: quick_action_tapped, Properties: {"button":"posts"}
🔵 Tracked: quick_action_tapped, Properties: {"button":"media"}
🔵 Tracked: quick_action_tapped, Properties: {"button":"pages"}
```


**Test B: Tabs enabled**
- Launch the app
- Navigate to Me -> App Settings -> Debug Settings
- Enable `MySiteDashboardTabsFeatureConfig` and restart the app
- Navigate to My Site View
- Make "HOME" the selected tab by tapping on it
- Tap on the following (returning back to My Site View after each click):  
   - All Quick Link buttons (Stats, Pages, Posts, and Media) 
-  Swipe down on the My Site View to request a pull-to-refresh
- View the logs and note that "tab_source=dashboard" is included in the properties of the track event.  
```
🔵 Tracked: my_site_pull_to_refresh, Properties: {"default_tab_experiment":"site_menu","tab_source":"dashboard"}
🔵 Tracked: quick_action_tapped, Properties: {"default_tab_experiment":"site_menu","tab_source":"dashboard","button":"stats"}
🔵 Tracked: quick_action_tapped, Properties: {"default_tab_experiment":"site_menu","tab_source":"dashboard","button":"posts"}
🔵 Tracked: quick_action_tapped, Properties: {"default_tab_experiment":"site_menu","tab_source":"dashboard","button":"media"}
🔵 Tracked: quick_action_tapped, Properties: {"default_tab_experiment":"site_menu","tab_source":"dashboard","button":"pages"}
```
- Make "SITE MENU" the selected tab by tapping on it
- Repeat the quick link taps/PTR request
-  View the logs and note that "tab_source=site_menu" is included in the properties of the track event.
```
🔵 Tracked: my_site_pull_to_refresh, Properties: {"default_tab_experiment":"site_menu","tab_source":"site_menu"}
🔵 Tracked: quick_action_tapped, Properties: {"default_tab_experiment":"site_menu","tab_source":"site_menu","button":"stats"}
🔵 Tracked: quick_action_tapped, Properties: {"default_tab_experiment":"site_menu","tab_source":"site_menu","button":"posts"}
🔵 Tracked: quick_action_tapped, Properties: {"default_tab_experiment":"site_menu","tab_source":"site_menu","button":"media"}
🔵 Tracked: quick_action_tapped, Properties: {"default_tab_experiment":"site_menu","tab_source":"site_menu","button":"pages"}
```
## Regression Notes
1. Potential unintended areas of impact
Events for pull-to-refresh, quick links are not tracked

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and automated testing

3. What automated tests I added (or what prevented me from doing so)
Add new tests to `MySiteViewModelTest`

PR submission checklist:
